shell/platform/windows/flutter_windows_texture_registrar.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/flutter_windows_texture_registrar.h"
+
+#include "flutter/shell/platform/windows/flutter_windows_engine.h"
+
+namespace flutter {
+
+FlutterWindowsTextureRegistrar::FlutterWindowsTextureRegistrar(
+    FlutterWindowsEngine* engine)
+    : engine_(engine) {}
+
+int64_t FlutterWindowsTextureRegistrar::RegisterTexture(
+    FlutterDesktopTextureCallback texture_callback,
+    void* user_data) {
+  auto texture_gl =
+      std::make_unique<flutter::ExternalTextureGL>(texture_callback, user_data);
+
+  int64_t texture_id = texture_gl->texture_id();
+  textures_[texture_id] = std::move(texture_gl);
+
+  if (FlutterEngineRegisterExternalTexture(engine_->engine(), texture_id) ==
+      kSuccess) {
+    return texture_id;
+  }
+
+  return -1;
+}
+
+bool FlutterWindowsTextureRegistrar::UnregisterTexture(int64_t texture_id) {
+  auto it = textures_.find(texture_id);
+
```

Comment:
Nit: The C++ style guide prefers using horizontal whitespace only to separate conceptually distinct code; this function is only really doing one thing, so doesn't need the whitespace (and certainly a find and a check of the find result should not be separated). Same with PopulateTexture below.